### DISCUSSION TITLE
Re-exporting all pinia stores from index file

### DISCUSSION THREAD

**Discussion by miguelrk** [vuejs/pinia#838] [Help and Questions] [answered]
When using pinia with multiple stores, it becomes somewhat cumbersome to import all these in e.g. components:
```javascript
import { useStore1 } from '@/stores/store-1'
import { useStore2 } from '@/stores/store-2'
import { useStore3 } from '@/stores/store-3'

export default {
  setup() {
    const store1 = useStore1()
    const store2 = useStore2()
    const store3 = useStore3()

    return { store1, store2, store3 }
  },
}
```

Is it possible to re-expose all stores from an index file e.g. `/stores/index.ts` **without loosing on code splitting** or should this be avoided?
```javascript
import { useStore1, useStore2, useStore3 } from '@/stores'

export default {
  setup() {
    const store1 = useStore1()
    const store2 = useStore2()
    const store3 = useStore3()

    return { store1, store2, store3 }
  },
}
```

**Comment by posva** [comment, marked as answer]
It should still work when it comes to codesplit as long as your `@/stores/index.ts` file is just doing `export * from './store-1'` for each store. But I would give it a try by using two different stores imported from there in two routes and see if the generated chunks correctl contain only one of them.

**Reply by miguelrk** [reply]
Thanks for the quick response @posva ! Exactly, the `@/stores/index.ts` is just doing `export * from './store-1'`. I'll try it out and post my findings anyways! But this gives me peace of mind already 😄
